ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -3562,8 +3546,28 @@ unsigned HqlCppTranslator::buildRtlField(StringBuffer * instanceName, IHqlExpres
             break;
         }
 
-        if (xpathName.length() && (xpathName.charAt(0) == '@'))
-            typeFlags |= RFTMhasxmlattr;
+        if (xpathName.length())
+        {
+            if (xpathName.charAt(0) == '@')
+                typeFlags |= RFTMhasxmlattr;
+            if (!strpbrk(xpathName, "/?*[]<>"))
+                typeFlags |= RFTMxpathscalar;
```

Comment:
This way around means that old workunits that have not been recompiled will not behave the same way.  This flag will not be set, which means they will now use the field name.  The type flag needs to be something like RFTMhasnonscalarxpath.